### PR TITLE
refactor, chore(egen): Removes boilerplate, heading fixes, and other corrections from template

### PR DESCRIPTION
**Pull request by nileshspringml**
**REQUIRED:** 

- Updated template of notebook
- Removed boilar code
- Changed REGION variable name to LOCATION
- Service account permission changed as we don't need admin level access for this notebook
- Added code for picking up correct service account

<br>
Removed boilar code and did required changes related to service account code
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/official/custom/custom_training_tensorboard_profiler.ipynb [hunk 1]
```diff
@@ -591,23 +536,22 @@
...
-      "attachments": {},
       "cell_type": "markdown",
       "metadata": {
         "id": "MPqbvhO6tK_e"
       },
       "source": [
         "### Configure authentication to your private Docker repository\n",
         "\n",
-        "Before you push or pull container images, configure Docker to use the `gcloud` command-line tool to authenticate requests to `Artifact Registry` for your region."
+        "Before you push or pull container images, configure Docker to use the `gcloud` command-line tool to authenticate requests to `Artifact Registry` for your location."
```
Comment (kittyabs) [inline notebooks/official/custom/custom_training_tensorboard_profiler.ipynb:554]: Artifact Registry references a product. Remove back ticks:

- `Artifact Registry` should change to: Artifact Registry

**Review by kittyabs — COMMENTED** [review]
one suggested edit

File: notebooks/official/custom/custom_training_tensorboard_profiler.ipynb (hunk 1 above)
Comment (nileshspringml) [reply notebooks/official/custom/custom_training_tensorboard_profiler.ipynb]: Did the required changes

**Review by katiemn — APPROVED** [review]

**Pull request merged**